wom/src/main/scala/wom/values/WomObject.scala

```diff
@@ -23,6 +27,18 @@ trait WomObjectLike extends WomValue {
   override def collectAsSeq[T <: WomValue](filterFn: PartialFunction[WomValue, T]): Seq[T] = {
     values.values.toSeq flatMap { _.collectAsSeq(filterFn) }
   }
+
+  def traverse[R <: WomValue, G[_]](f: WomValue => G[R])(implicit applicative: Applicative[G]): G[WomObjectLike] = {
+    if (values.isEmpty) applicative.pure(this)
+    else {
+      val traverseFunction: (String, WomValue) => G[(String, R)] = {
+        case (key, value) => applicative.map(f(value)) { key -> _ }
+      }
+      applicative.map(values.toList.traverse[G, (String, R)](traverseFunction.tupled)) { mapped =>
+        copyWith(mapped.toMap)
```

Comment:
ToL: Uh... I've got lot of FP homework to do.

Reply:
There might be a simpler way to write this tbh